src/queries/rum-pageviews-pivot.sql

```diff
@@ -0,0 +1,110 @@
+--- description: Get monthly page views for a site according to Helix RUM data
+--- Authorization: none
+--- Access-Control-Allow-Origin: *
+--- interval: 365
+--- offset: 0
+--- url: -
+--- timezone: UTC
+--- device: all
+--- domainkey: secret
+
+-- prepare two strings to insert into query below
+-- pivot command requires specific values, not dynamic results
+DECLARE months12 STRING;
+DECLARE months3 STRING;
+
+SET months12 = (
+  SELECT
+    CONCAT(
+      '("',
+      STRING_AGG(DISTINCT FORMAT_DATE('%Y-%b', alldays), '", "'),
+      '")'
+    )
+  FROM
+    UNNEST(
+      GENERATE_DATE_ARRAY(
+        CURRENT_DATE(@timezone) - 365,
+        CURRENT_DATE(@timezone) - 1
+      )
+    ) AS alldays
+  WHERE FORMAT_DATE('%d', alldays) = '01'
+);
+SET months3 = (
+  SELECT
+    CONCAT(
+      '("',
+      STRING_AGG(DISTINCT FORMAT_DATE('%Y-%b', alldays), '", "'),
+      '")'
+    )
+  FROM
+    UNNEST(
+      GENERATE_DATE_ARRAY(
+        CURRENT_DATE(@timezone) - 90,
+        CURRENT_DATE(@timezone) - 1
+      )
+    ) AS alldays
+);
+
+-- workaround to put dynamic list of specific values into PIVOT command
+-- otherwise month/year would need to be hardcoded
+EXECUTE IMMEDIATE format( -- noqa: PRS
+"""
+WITH pvs AS (
+  SELECT
+    SUM(pageviews) AS pageviews,
+    REGEXP_REPLACE(hostname, r'^www.', '') AS hostname,
+    FORMAT_DATE('%%Y-%%b', time) AS month
+  FROM
+    helix_rum.PAGEVIEWS_V3(
+      '%s',
+      %s,
+      %s,
+      '',
+      '',
+      '%s',
+      '%s',
+      '%s'
+    )
+  WHERE
+    hostname != ''
+    AND NOT REGEXP_CONTAINS(hostname, r'^\\d+\\.\\d+\\.\\d+\\.\\d+$') -- IP addresses
+    AND hostname NOT LIKE 'localhost%%'
+    AND hostname NOT LIKE '%%.hlx.page'
+    AND hostname NOT LIKE '%%.hlx3.page'
+    AND hostname NOT LIKE '%%.hlx.live'
+    AND hostname NOT LIKE '%%.helix3.dev'
+    AND hostname NOT LIKE '%%.sharepoint.com'
+    AND hostname NOT LIKE '%%.google.com'
+    AND hostname NOT LIKE '%%.edison.pfizer' -- not live
+    AND hostname NOT LIKE '%%.web.pfizer'
```

Comment:
This should be handled through domain key validity rather than hardcoding it

Reply:
The challenge here is not about domain key validity, but separating development from live domains (especially for the global domain keys).  I believe that contractually we only count production traffic.  In any case point taken about hardcoding, I've been contemplating a domain exclusion/tagging table and will revisit this query in the event we move in that direction.  Such a table could contain metadata about domains that we could use for various purposes in our queries.

Reply:
Good idea on the temp table, this is done and looks cleaner.